### PR TITLE
Add asociations

### PR DESCRIPTION
**Pull request by ShoiraTa**
## In this branch I :

- Added all the required classes
- Set up associations between the classes

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative 'rental'
+# Book class
+class Book
+  attr_accessor :title, :author
```
Comment (julie-ify) [inline book.rb]: - [ ] Let's add `rentals` to `attr_accessor` as user can read and write rentals too.

```suggestion
class Book
  attr_accessor :title, :author, :rentals
```

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(_students)
+    @add_student << student
+    student.classroom = self
+  end
```
Comment (julie-ify) [inline classroom.rb]: - [ ] Let's define the `add_student` method properly by making sure the argument passed to it is consistent and is being used inside the method. If you noticed, the argument you're passing is `_students` but there is no place that you're making use of this argument, instead, you're making use of `student` which is not defined anywhere in the method. Let's fix this by modifying the method to accept and make use of the argument and to push the argument to the `students` array defined in the `initialize` method. See my suggestion below;

```suggestion
  def add_student(student)
    @students << student
    student.classroom = self
  end
```

File: person_class.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 require './corrector'
+require_relative 'rental'
 # Person class
 class Person
   attr_reader :id
-  attr_accessor :name, :age, :parent_permission
+  attr_accessor :name, :age, :parent_permission, :rental
```
Comment (julie-ify) [inline person_class.rb]: - [ ] Let's correct the typo error here as we are to add `:rentals` to `attr_accessor` and not `:rental`

File: person_class.rb [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 
   def initialize(age, parent_permission = 'true', name = 'Unknown')
     @id = Random.rand(1..500)
```
Comment (julie-ify) [inline person_class.rb:9]: - [ ] To fulfill the `has_many` relationship between Person and Rental, remember that a person can have many rentals. Let's add `rentals` to the initialized method. Add this `@rentals = []` to the `initialize` method. This is also why I suggested that you add `rentals` to the `attr_accessor` 😉.

File: student_class.rb [hunk 5]
```diff
@@ -1,13 +1,17 @@
 require './person_class'
+require_relative 'classroom'
 # Student class
 class Student < Person
-  attr_accessor :classroom
```
Comment (julie-ify) [inline student_class.rb:4]: - [ ] Let's ensure that we can be able to read the `classroom` by adding it to attribute reader. 

`attr_reader :classroom`

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @ShoiraTa,

Good job so far!

## Highlight!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️  
- Good code organization ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

- [ ] **[OPTIONAL]** Let's improve the README by having the correct instructions there. Let's remove this line `Go to the dist folder and open index.html` as there is no dist folder and index.html file in ruby projects. 😉.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 2 above)
Comment (ShoiraTa) [reply classroom.rb]: Thank you for detailed comment, I fixed this

**Comment by ShoiraTa** [comment]
@julie-ify I highly appreciate your time and efforts  reviewing my work 💯 . I have fixed all the mentioned issues.

File: classroom.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @add_student << student
+    student.classroom = self
+  end
```
Comment (julie-ify) [inline classroom.rb]: - [ ] Let's push the student received as an argument to the `@students` instance variable which is an array defined in the initialize method. Right now you're pushing to `@add_student` instance variable which does not exist. See suggestion below;

```suggestion
  def add_student(student)
    @students << student
    student.classroom = self
  end
```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @ShoiraTa, 

Good job so far!

## Highlight!
- Good job implementing some of the suggested changes from the previous review ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @ShoiraTa,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**